libraries/chain/block_log.cpp

```diff
@@ -643,6 +700,10 @@ namespace eosio { namespace chain {
 
    block_id_type block_log::read_block_id_by_num(uint32_t block_num)const {
       try {
+         if (my->not_generate_block_log) {
+            ilog("Not possible to read_block_id_by_num in no blocks.log mode (block-log-retain-blocks=0)");
```

Comment:
All these "error" ilogs seem rather chatty. Consider just return nulls or asserting if they should never be called in no block log mode.